src/nvim/syntax.c

```diff
@@ -7760,6 +7760,17 @@ static void highlight_list_two(int cnt, int attr)
 const char *get_highlight_name(expand_T *const xp, const int idx)
   FUNC_ATTR_WARN_UNUSED_RESULT
 {
+  if (idx < 0) {
+    return NULL;
+  }
+
+  // Items are never removed from the table, skip the ones that were cleared.
+  int current_idx = idx;
```

Comment:
`current_idx` was added in https://github.com/neovim/neovim/pull/7424 . We need to remove it now. And we need to remove `const` from the `idx` parameter.